File: Dart/degiskenler2.dart
main(){ // @raifpy
  // var ve tür olarak tanımlamayı biliyorduk zaten

  var isim = "Merhaba";
  String isim2 = "Dünya";

  print(isim+" | "+isim2);

  // Peki isim 'i (var-String) başka bir tür yapmak istersek ne olur ?

  isim = "Hey"; // Bunda elbette problem yok .

  // isim = true; // Hata : Error: A value of type 'bool' can't be assigned to a variable of type 'String'
  
  // Aynı şekilde isim2^de hata verecekti .


  dynamic nesne = "Merhaba Dost .";
  print(nesne); 

  nesne = true;

  print(nesne); //

  // ama var yerine dynamic yapısını kullanarak tür değişikliği yaparsak herhangi bir problem olmuyor :)

  // 

  // dynamic 
  // var
  // tür
   // Python değişkenleri dynamic yapısını kullanıyormuş .




}

File: Go/degiskenler2.go
package main

import "fmt"

func main() {
	var _1 = "Merhaba"
	_2 := "Dünya"
	fmt.Println(_1, _2)  // , yapılarını kullandım
	fmt.Println(_1 + _2) // + yapısını kullandım

	//fmt.Println(_1 * 10) // Hata veriyor .
	//fmt.Println(_1 - _2) // Hata veriyor .

	_3 := 10
	var _4 = 20

	fmt.Println(_3 + _4) // Toplama işlemi yapmış olduk (+)
	fmt.Println(_3, _4)  // , kullandığimız için sadece yazdı

	//fmt.Println(_2+_3) // string and int hata verdi (doğal olarak)
	fmt.Println(_2, _3) //

	// Python yapısı ile aynı |

}
